proposals/001_interop.md

```diff
@@ -0,0 +1,473 @@
+head may be per user.
+
+This endpoint MUST handle parameters supplied as URL Query Parameters, and SHOULD NOT support accepting these
+parameters as fields in the request body.
+
+- from: a sync head. if present, must be number, can be timestamp or preferably a serial. if omitted, start from earliest known.
+Conventionally, element(s) that match `from` are included. Timestamp implementations of `from` may have unspecified behavior
+and are not recommended.
+- limit: a specified maximum number of elements. Because Docmaps data are natively graph based, the semantics of this
+limit are left to implementers and SHOULD NOT be relied upon by clients to produce predictable behavior. A given server
+MAY encourage consumers of the API to use this parameter to intentfully specify their capacity for pagination. A client
+MUST NOT assume that all servers will handle this parameter in the same way.
+- session: OPTIONAL, any string identifier that the server may require for purposes of tracking session-based syncs. The server
+may decide whether this field is required and any further substructure expected within the string that their session cardinality
+may depend on. Note that session identification relates  to client identification  and therefore there are security considerations.
+For example, a server may issue a session token on request for a sync with no `from` specified, as starting a new sync
+session. This session token can be used to disrupt the sync session and so is considered a secret, even though the impact of
+impersonation may not be severe.
+
+**Response** MUST respond with an LD-graph  of objects that may be disjoint. These objects MUST all have
+IRIs in their `ID` field.
+
+Note that this endpoint is asssumed to be dependent on pagination, and MUST conform to LDP Paging (see above).
+
+The `next` `Link` response header is REQUIRED for this endpoint. If that link refers to a next page which does not yet exist,
+because the synchronizing client has caught up, it MUST serve `HTTP 202 Accepted` response. Clients are expected to resubmit
+the same query later according to a reasonable back-off scheme.
+
+```http
+GET https://example.com/docmaps/v1/synchronization?from=9999&limit=9999&session=ff000000
+
+HTTP/1.1 200 OK
+Content-Type:  application/ld+json
+Link: <https://example.com/docmaps/v1/synchronization?from=19998&limit=9999&session=ff000000>; rel="next"
+```
+
+```json
+{
+   "@context": "https://w3id.org/docmaps/context.jsonld",
+   "@graph": [
+      {
+        "type": "docmap",
+        "id": "https://example.com/ex/docmaps/v1/docmap/deadbeef-f1e7-4b2c-a6f5-d8c2f9f9a2e2"
+      }
+   ]
+}
+```
+
+where
+
+| Field | Value |
+| -------- | -------- |
+| `@context` |  If present, MUST be a JSON-LD context. SHOULD be exactly `"https://w3id.org/docmaps/context.jsonld"`  |
+| `@graph` | MUST be a json-ld graph. |
+
+By recommendation of the authors, this endpoint SHOULD NOT respond with elements of `"type": "docmap"`, though this is
```

Comment:
I am not sure whether I am reading this correctly. Isn't the example responding with the type docmap?

Reply:
Yes, I have a note to self to clarify this. Thanks for calling it out.

The issue is that in theory, docmaps may be ephemeral/blank nodes/have names like "The docmap for DOI 10.1101/xyz as of this date" whereas things like Actions will be more stable. So I am pushing for persistent entities to be the core response of sync actions.

However, for everyone as an ecosystem to move forward on this paradigm is blocked on those persistent objects having an updatedAt field and a persistent ID (`IRI`).